[ci] perform test selection outside container

aslonnie opened the pull request:
so that it does not run inside the test container. the logic is mostly bazel query to do test selection, there is no need to launch a container to run it.

aslonnie commented:
needs non-root forge first..

elliot-barn reviewed (changes requested):
Can you update the unit tests in `test_tester.py`

aslonnie committed:
[ci] perform test selection outside container

they do not (and should not) happen inside the test container;
bazel query can run anywhere.

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
